kyo-prelude/shared/src/test/scala/kyo/GuardTest.scala

```diff
@@ -0,0 +1,95 @@
+package kyo
+
+class GuardTest extends Test:
```

Comment:
Can you add tests with multiple effects? It'd be nice to have scenarios where only part of the effects are guarded

Reply:
This approach does not support masking specific effects within a larger effect intersection. I added a test with multiple effects though.